app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,25 @@
+class RecipesController < ApplicationController
+  def index
+    @recipes = Recipe.all.where(user: current_user)
+  end
+
+  def show
+    @recipe = Recipe.find(params[:id])
+    @recipes_food = RecipeFood.all.where(recipe: @recipe)
+  end
+
+  def destroy
+    @recipe = Recipe.find(params[:id]).destroy!
+    redirect_to recipes_path, notice: 'Recipe has been deleted successfully!'
+  end
+
+  def toggle
+    @recipe = Recipe.find(params[:id])
+    @recipe.toggle!(:public)
+    redirect_to recipes_detail_path(params[:id])
+  end
+
+  def public_recipes
+    @recipes = Recipe.all.where(public: true)
+  end
+end
```

Comment:
- [ ] Please note that the recipe & public recipe pages are returning an empty page with no button to create a recipe, kindly fix that issue by adding a link to create a recipe and make sure it matches the wireframe and has user authorization rules 🙏 


![image](https://user-images.githubusercontent.com/89282221/230735179-ee2ea544-6bcd-4fc1-bd8f-4ddfcc567f9a.png)